### ISSUE TITLE
Malformed server configuration

### ISSUE THREAD

**Issue by micah** [nextcloud/android#2227]
I'm able to connect my android device to my nextcloud server. It logs in, and is able to load the top-level folder list. I can view the files on the server.

If I try to make a new folder there, or upload files, I am given the error, "Malformed server configuration". 

I can use the web interface, and create folders, and upload files, without any problem.

http://192.168.1.45/remote.php/webdav brings up a htpasswd login box, if I login there, I see:

This is the WebDAV interface. It can only be accessed by WebDAV clients such as the Nextcloud desktop sync client.

http://192.168.1.45/status.php shows me:

```
{"installed":true,"maintenance":false,"needsDbUpgrade":false,"version":"13.0.0.14","versionstring":"13.0.0","edition":"","productname":"Nextcloud"}
```

### Expected behaviour

It should properly create folders and files

### Environment data
Android version: 7.1.2

Device model: SM-T700

Stock or customized system: LineageOS

Nextcloud app version: 3.0.1

Nextcloud server version: 13.0

#### Web server error log
```
192.168.1.47 - micah [24/Feb/2018:09:48:56 -0500] "MKCOL /remote.php/webdav/Foo/ HTTP/1.1" 500 211 "-" "Mozilla/5.0 (Android) ownCloud-android/3.0.1"
```

#### Nextcloud log (data/nextcloud.log)
I'm having trouble finding this file

**Comment by tigernero79** [comment]
in the config.php file under 

'trusted_domains' =>
  array (
    0 => 'www.example.com,
    1 => '192.168.1.45',
  ), 
did you enter local ip is public ip?

**Comment by micah** [comment]
tigernero79 <notifications@github.com> writes:

> in the config.php file under 
>
> 'trusted_domains' =>
>   array (
>     0 => 'www.example.com,
>     1 => '192.168.1.45',
>   ), 
> did you enter local ip is public ip?

no, my nextcloud server is on a local, non-public network.

**Comment by AlexKalopsia** [comment]
Hi, I am facing the same exact issue. Can perfectly reach my NC via browser and upload/download with no issue. But when I do this via my Android phone, I get the "Malformed server configuration" problem

**Comment by thebitstick** [comment]
I am now facing this issue as well. NC Android reported "Malformed server configuration". NC Web works just fine. NC via WebDAV works just fine.
I have touched no configuration files in months, although the Snap I installed NC from has updated a few times.

**Comment by mpgoebel** [comment]
This started happening for me today as well.

**Comment by mjgiardino** [comment]
I've also begun to get this. Autoupload seems to work generally, it's large videos that fail, but with no obvious failure log. Here is access.log for the attempts

`IP - - [02/Apr/2020:12:33:10 +0200] "GET /index.php/204 HTTP/1.1" 204 1370 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.11.0"
IP - - [02/Apr/2020:12:33:10 +0200] "GET /index.php/204 HTTP/1.1" 204 1364 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.11.0"
IP - - [02/Apr/2020:12:33:11 +0200] "GET /index.php/204 HTTP/1.1" 204 1364 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.11.0"
IP - user [02/Apr/2020:12:33:11 +0200] "HEAD /remote.php/webdav/Photos/Camera%20Uploads/2020/04/VID_20200401_200509.mp4 HTTP/1.1" 404 729 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.11.0"
IP - user [02/Apr/2020:12:33:11 +0200] "MKCOL /remote.php/dav/uploads/giardino/78b5d53b59e3b75cfe5100fda851e6cd HTTP/1.1" 405 954 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.11.0"
IP - user [02/Apr/2020:12:33:11 +0200] "PROPFIND /remote.php/dav/uploads/giardino/78b5d53b59e3b75cfe5100fda851e6cd HTTP/1.1" 500 1208 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.11.0"`

**Comment by RobTranquillo** [comment]
I try to connect my  Threema Safe Backup (no large files involved, just a connectivity test) with the error message "malformed server response" on my NC 18.0.3.  Connecting via Windows Explorer work fine. Curl in windows says:

> <?xml version="1.0" encoding="utf-8"?>                                                                                                                                                      <d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">                                                                                                                                     <s:exception>Sabre\DAV\Exception\NotAuthenticated</s:exception>                                                                                                                             <s:message>No public access to this resource., No 'Authorization: Bearer' header found. Either the client didn't send one, or the server is mis-configured, No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured</s:message>                                                                                                     </d:error>  

access.log:

> xxx.xxx.xxx.xxx - root [02/Apr/2020:12:52:02 +0000] "GET /nextcloud/remote.php/dav/files/root/Dokumente/threema_safe_rob/config HTTP/1.0" 200 1565 "-" "Threema"

**Comment by Nebucatnetzer** [comment]
I'm getting the same error with a Docker setup.

**Comment by owenashurst** [comment]
Also getting the issue as of today. Have Nextcloud installed in Docker also.

**Comment by f4grx** [comment]
Same error as everyone else. manual install on a raspi, I can use generally, but auto upload fails and the android client shows this malformed server configuration (Configuration Serveur Erronée in french). Failed uploads end up in the "try again zone" with unknown errors.

**Comment by AlexKalopsia** [comment]
Oh wow, I thought this was just me, hopefully someone from the team can look into this soon

**Comment by thebitstick** [comment]
Update: I can no longer reproduce this as my Nextcloud instance on Snap auto-updated.

**Comment by f4grx** [comment]
Note: I could get my failed files sent by RE-sending them manually, one by one. But the auto-upload was no happy with the existing entries, that kept failing in a loop. I had to remove them from the list.

The Malformed server configuration is still present but did not prevent upload. I susped there are different issues here.

**Comment by owenashurst** [comment]
I can confirm manually uploading works but auto upload complains of the
Malformed Server Configuration.

**Comment by Goeny** [comment]
Also having the same problem. Malformed server configuration on Android
Nextcloud 18.0.3
PHP 7.3.13
CentOS 7

I just recently updated NC to v18.0.3
Not sure but have the feeling it started after that update.

**Comment by DrRSatzteil** [comment]
I can confirm upload issues: some files are successfully uploaded while others keep failing with "Unbekannter Fehler" (unknown error). The notification says "Fehlerhafte Serverkonfiguration" (Malformed server configuration)

NC 18.0.3
PHP 7.3
Apache2
Ubuntu 18.04

**Comment by benedetto** [comment]
I have the same. See my comment here: https://github.com/nextcloud/android/issues/5459#issuecomment-612670096

**Comment by tobiasKaminsky** [comment]
Let us track this all in #5459

**Comment by danwize** [comment]
I was having this issue too.  My nextcloud has an external smb share and the folder that I was uploading to was owned by www-data:www-data.  This was working when nextcloud was running on my NAS as www-data user, but I moved to a new instance and mounted a SMB share with the user dan.  I ran chown dan:dan /mnt/path-to-my-problem-folder and I am now able to upload files to that folder.  Auto upload and manual upload are working now.
